packages/headless/src/features/history/history-slice.ts

```diff
@@ -152,6 +155,32 @@ const isCategoryFacetsEqual = (
   return true;
 };
 
+const isAutomaticFacetsEqual = (
+  current: AutomaticFacetSetState,
+  next: AutomaticFacetSetState
+) => {
+  for (const [key, value] of Object.entries(next.set)) {
+    if (!current.set[key]) {
+      return false;
+    }
+
+    const currentSelectedValues = (
+      current.set[key].response.values as FacetValue[]
+    ).filter((value) => value.state === 'selected');
```

Comment:
Automatic facets will likely support facet exclusion in the future, so might as well save ourselves the hassle now 😅 

```suggestion
    ).filter((value) => value.state !== 'idle');
```